src/manifest/manifest.ts

```diff
@@ -366,6 +366,19 @@ export default class Manifest extends EventEmitter<IManifestEvents> {
     });
   }
 
+  /**
+   * Returns the Period encountered at the given time if one, or the next Period
+   * if that time is between two Periods.
+   * Returns `undefined` if there is no Period at or after that time.
+   * @param {number} time
+   * @returns {Object|undefined}
+   */
+  public getNextPeriod(time : number) : Period | undefined {
```

Comment:
I am not a huge fan of calling it that way if it returns most of the time (I guess ?) the period for the given time.
Wouldn't it be clearer to still have the `getPeriodForTime` and `getNextPeriod` (that really gives you the first period that starts after the given time) ?

Reply:
I've thought about it but this would mean both a more complex `getNextPeriod` implementation (because we would have to check that this is not the current one) and most likely always write the following when using it:
```ts
const period = getPeriodForTime(time) ?? getNextPeriod(time);
if (period === null) {
  // do something
}
```
Which I find less elegant.

Maybe we could think of a better name for that method?

Reply:
Maybe it's less elegant but it's much more intelligible. And isn't it what matters ?

Otherwise, maybe we could think of something closer to `getFirstPeriodThatEndsAfterTime` (🥴)

Reply:
done